src/shared/AutoApprovalSettings.ts

```diff
@@ -3,8 +3,10 @@ export interface AutoApprovalSettings {
 	enabled: boolean
 	// Individual action permissions
 	actions: {
-		readFiles: boolean // Read files and directories
```

Comment:
since the object will already exist in globalStorage, these values will still exist and the new readFilesLocally will be defaulted to non-existant (instead of false), so:
- we need to set new properites as optional
- this may disrupt existing users that have enabled readFiles/editFiles if we are not migrating them over
i think the best solution here is that we maintain the original readFiles and editFiles properties as they are, and add new optional properties for readFilesExternally and editFilesExternally. Alternatively we could do a bit of migration logic in `getAllExtensionState` if we want to maintain the `readFilesLocally`/`editFilesLocally` names.